tools/editor/editor_settings.cpp

```diff
@@ -644,6 +659,127 @@ void EditorSettings::load_favorites() {
 
 }
 
+void EditorSettings::list_text_editor_themes() {
+	String themes="Default";
+	DirAccess *d = DirAccess::open(settings_path + "/text_editor_themes");
+	if (d) {
+		d->list_dir_begin();
+		String file = d->get_next();
+		while(file != String()) {
+			if (file.extension() == "tet" && file.basename().to_lower() != "default") {
+				themes += "," + file.basename();
+			}
+			file = d->get_next();
+		}
+		d->list_dir_end();
+		memdelete(d);
+	}
+	add_property_hint(PropertyInfo(Variant::STRING,"text_editor/color_theme",PROPERTY_HINT_ENUM,themes));
+}
+
+void EditorSettings::load_text_editor_theme() {
+	if (get("text_editor/color_theme") == "Default") {
+		_load_default_text_editor_theme();	// sorry for "Settings changed" console spam
+		return;
+	}
+
+	String theme_path = get_settings_path() + "/text_editor_themes/" + get("text_editor/color_theme") + ".tet";
+
+	Ref<ConfigFile> cf = memnew( ConfigFile );
+	Error err = cf->load(theme_path);
+
+	if (err != OK) {
+		return;
+	}
+
+	List<String> keys;
+	cf->get_section_keys("color_theme", &keys);
+
+	for(List<String>::Element *E=keys.front();E;E=E->next()) {
+		String key = E->get();
+		String val = cf->get_value("color_theme", key);
+
+		// don't load if it's not already there!
+		if (EditorSettings::get_singleton()->has("text_editor/" + key)) {
```

Comment:
`EditorSettings::get_singleton()` is not needed here. I think there is another one below.